### PR TITLE
Happy Gains Beside Candy

### LINKED ISSUE #109: Display total Happy Gains beside name of the Candy

### PR DESCRIPTION
**Pull request by Sashank999**
Closes #109 .

### CODE REVIEW

File: extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+					torndata.items[candy.dataset.item].effect
+						.split(" ")
+						.map((x) => parseInt(x))
+						.filter((x) => !isNaN(x))[0]
+				);
+				let totalHappy = baseHappy;
+				if (!isNaN(facCandyPerc)) totalHappy += (facCandyPerc / 100) * baseHappy;
+				if (!isNaN(jobCandyPerc)) totalHappy += (jobCandyPerc / 100) * baseHappy;
+				const rawHTML = `<span class="tt-candy-gains">${totalHappy}H</span>`;
+				candy.find(".name-wrap .qty.bold.t-hide").insertAdjacentHTML("beforeEnd", rawHTML);
```
Comment (DeKleineKobini) [inline extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js:52]: Starting in v6, I'll also make a point of "beforeEnd" etc.

According to [the docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML), it's "beforeend", without the uppercase.

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]

File: extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		{
+			storage: ["settings.pages.items.candyHappyGains"],
+		},
+		async () => {
+			await requireElement("[data-category='Candy']");
+		}
+	);
+
+	function initialiseAddGains() {
+		CUSTOM_LISTENERS[EVENT_CHANNELS.ITEM_SWITCH_TAB].push(() => {
```
[the remark this answers is not in the document]
Comment (Sashank999) [reply extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js:20]: For ITEMS_LOADED, I already have it in requirements.
It is to monitor when user switches the categories by clicking icons.

File: extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js (hunk 1 above)
Comment (Sashank999) [reply extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js:52]: It was first in camelCase, like I typed.
I think the standard is changed. For a reference, it is here http://help.dottoro.com/ljbreokf.php
I will change it.

Comment (Sashank999) [reply extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js]: Changed.

File: extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (DeKleineKobini) [reply extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js:20]: ITEMS_LOADED also is used for scrolling

[the remark this answers is not in the document]
Comment (DeKleineKobini) [reply extension/scripts/features/happygainsbesidecandy/ttHappyGainsBesideCandy.js:20]: Screenshot

**Comment by Sashank999** [comment]
Done.

**Review by DeKleineKobini — APPROVED** [review]

**Pull request merged**